### PR TITLE
GDB-4541 Import JSON mapping definition in UI

### PR DESCRIPTION
**Pull request by teodossidossev**

### CODE REVIEW

File: cypress/test/main/mapper/header-component.spec.ts [hunk 1]
```diff
@@ -1,14 +1,94 @@
...
+    MappingSteps.getConfirmationMessage().should('contain', 'All mappings will be overwritten. Do you want to proceed?');
+    // I confirm
+    MappingSteps.confirm();
+    // THEN:
+    // A new mapping is loaded
+    cy.cypressData(MapperComponentSelectors.SUBJECT_SELECTOR + "-0").should('be.visible').and('contain', 'Pesho');
+    cy.cypressData(MapperComponentSelectors.PREDICATE_SELECTOR + "-0").should('be.visible').and('contain', 'loves');
+    cy.cypressData(MapperComponentSelectors.OBJECT_SELECTOR + "-0").should('be.visible').and('contain', 'Maria');
+
+    cy.cypressData(MapperComponentSelectors.SUBJECT_SELECTOR + "-4").should('be.visible').and('contain', '');
```
Comment (svilenvelikov) [inline cypress/test/main/mapper/header-component.spec.ts:50]: umm no, this is not a good check, better use `and('should.be.empty')` or `and('should.have.length')`

File: cypress/test/main/mapper/header-component.spec.ts [hunk 2]
```diff
@@ -1,14 +1,94 @@
...
-    // GIVEN:
-    // I visit home page
-    cy.visit('');
     // THEN:
     // I see header content
     cy.cypressData(HeaderComponentSelectors.HEADER_SELECTOR).should('be.visible');
   });
+
+  it('should upload JSON mapping', () => {
+    cy.visit('?dataProviderID=ontorefine:123');
```
Comment (svilenvelikov) [inline cypress/test/main/mapper/header-component.spec.ts:29]: you already visited the page in the beforeEach

File: cypress/test/main/mapper/header-component.spec.ts [hunk 3]
```diff
@@ -1,14 +1,94 @@
...
+
+  it('should not insert uploaded JSON mapping if not confirmed', () => {
+    cy.visit('?dataProviderID=ontorefine:123');
+    // THEN:
+    // I see header content
+    cy.cypressData(MapperComponentSelectors.MAPPER_SELECTOR).should('be.visible');
+    // I see mapping holders
+    cy.cypressData(MapperComponentSelectors.SUBJECT_SELECTOR + "-0").should('be.visible').and('contain', 'director_name');
+    cy.cypressData(MapperComponentSelectors.PREDICATE_SELECTOR + "-0").should('be.visible').and('contain', 'test');
+    cy.cypressData(MapperComponentSelectors.OBJECT_SELECTOR + "-0").should('be.visible').and('contain', ' movie_imdb_link');
```
Comment (svilenvelikov) [inline cypress/test/main/mapper/header-component.spec.ts:63]: All the checks above are redundant as you've already verified them in previos test. Better remove them if you don't need them to give the cypress some time to catch up :)

File: cypress/test/main/mapper/header-component.spec.ts [hunk 4]
```diff
@@ -1,14 +1,94 @@
...
+
+  it('should display error message if file is wrong or JSON is corrupted', () => {
+    cy.visit('?dataProviderID=ontorefine:123');
+    // THEN:
+    // I see header content
+    cy.cypressData(MapperComponentSelectors.MAPPER_SELECTOR).should('be.visible');
+    // I see mapping holders
+    cy.cypressData(MapperComponentSelectors.SUBJECT_SELECTOR + "-0").should('be.visible').and('contain', 'director_name');
+    cy.cypressData(MapperComponentSelectors.PREDICATE_SELECTOR + "-0").should('be.visible').and('contain', 'test');
+    cy.cypressData(MapperComponentSelectors.OBJECT_SELECTOR + "-0").should('be.visible').and('contain', ' movie_imdb_link');
```
Comment (svilenvelikov) [inline cypress/test/main/mapper/header-component.spec.ts:86]: redundant checks above

File: src/app/main/mapper/header/header.component.ts [hunk 5]
```diff
@@ -76,4 +80,47 @@ export class HeaderComponent extends OnDestroyMixin implements OnInit {
...
+  }
+
+  private checkMappingValidity(newMapping) {
+    const mapping = classToClass(newMapping);
+    this.modelManagementService.removePreview(mapping);
+    Convert.mappingDefinitionToJson(mapping);
+  }
+
+  private showErrorWarning(message?: any) {
+    this.dialogService.error({
```
Comment (svilenvelikov) [inline src/app/main/mapper/header/header.component.ts:122]: Why not a toastr instead of this dialog? @desislava-hristova-ontotext

Comment (desislava-hristova-ontotext) [reply src/app/main/mapper/header/header.component.ts:126]: Yes, I guess a toast is better.

Comment (teodossidossev) [reply src/app/main/mapper/header/header.component.ts]: I replaced it with toast

**Pull request merged**